tools/StatelessExecution/SetupCli.cs

```diff
@@ -0,0 +1,177 @@
+            {
+                return Task.FromResult(1);
+            }
+
+            return Task.FromResult(0);
+        });
+    }
+
+    private static (Witness witness, Block block, BlockHeader parent) ReadData(string blockFileName, string witnessFileName)
+    {
+        var serializer = new EthereumJsonSerializer();
+        var witnessBytes = File.ReadAllText(witnessFileName);
+        var blockBytes = File.ReadAllText(blockFileName);
+
+        Witness witness = serializer.Deserialize<Witness>(witnessBytes);
+        BlockForRpc suggestedBlockForRpc = serializer.Deserialize<BlockForRpc>(blockBytes);
+
+        BlockHeader suggestedBlockHeader = new BlockHeader(
+            suggestedBlockForRpc.ParentHash,
+            suggestedBlockForRpc.Sha3Uncles,
+            suggestedBlockForRpc.Miner,
+            suggestedBlockForRpc.Difficulty,
+            suggestedBlockForRpc.Number!.Value,
+            suggestedBlockForRpc.GasLimit,
+            (ulong)suggestedBlockForRpc.Timestamp,
+            suggestedBlockForRpc.ExtraData,
+            suggestedBlockForRpc.BlobGasUsed,
+            suggestedBlockForRpc.ExcessBlobGas,
+            suggestedBlockForRpc.ParentBeaconBlockRoot,
+            suggestedBlockForRpc.RequestsHash)
+        {
+            StateRoot = suggestedBlockForRpc.StateRoot,
+            TxRoot = suggestedBlockForRpc.TransactionsRoot,
+            ReceiptsRoot = suggestedBlockForRpc.ReceiptsRoot,
+            Bloom = suggestedBlockForRpc.LogsBloom,
+            GasUsed = suggestedBlockForRpc.GasUsed,
+            MixHash = suggestedBlockForRpc.MixHash,
+            BaseFeePerGas = suggestedBlockForRpc.BaseFeePerGas!.Value,
+            WithdrawalsRoot = suggestedBlockForRpc.WithdrawalsRoot,
+            ParentBeaconBlockRoot = suggestedBlockForRpc.ParentBeaconBlockRoot,
+            RequestsHash = suggestedBlockForRpc.RequestsHash,
+            BlobGasUsed = suggestedBlockForRpc.BlobGasUsed,
+            ExcessBlobGas = suggestedBlockForRpc.ExcessBlobGas,
+            Hash = suggestedBlockForRpc.Hash,
+        };
+
+        BlockHeader? baseBlock = witness.DecodedHeaders.FirstOrDefault(h => h.Hash == suggestedBlockHeader.ParentHash);
+        if (baseBlock is null)
+        {
+            throw new ArgumentException($"No base block header in witness. Expected header with hash {suggestedBlockHeader.ParentHash}");
+        }
+
+        var transactions = new Transaction[suggestedBlockForRpc.Transactions.Length];
+        for (int j = 0; j < transactions.Length; j++)
+        {
+            var tx = (JsonElement)suggestedBlockForRpc.Transactions[j];
+            transactions[j] = serializer.Deserialize<TransactionForRpc>(tx.GetRawText()).ToTransaction();
+        }
+
+        Block suggestedBlock = new Block(suggestedBlockHeader, transactions, [], suggestedBlockForRpc.Withdrawals);
+        return (witness, suggestedBlock, baseBlock);
+    }
```

Comment:
All this decoding looks potentially heavy for zkVM context?
I wonder is it something we can improve in the future.

Do zkVM's use json data sources?